app/components/Pages/CategoryPage/index.js

```diff
@@ -0,0 +1,56 @@
+import { useState } from 'react';
+import MovieCard from 'components/UI/MovieCard';
+import {
+  WrapperGrid,
+  MovieCardWrapper,
+  SearchDiv,
+  SearchStyle,
+  ErrorWrapper,
+  NoMovieError,
+} from './CategoryPage.styles';
+import { Input } from 'components/Form';
+
+export default function CategoryPage({ moviesData, ...props }) {
+  if (moviesData.length === 0) {
+    return (
+      <ErrorWrapper>
+        <NoMovieError>Oops, looks like there are no movies in this category! :(</NoMovieError>
+      </ErrorWrapper>
+    );
+  }
+
+  const [searchQ, setSearchQ] = useState('');
+
+  const filteredList = searchQ
+    ? moviesData.filter((movie) => movie.title.includes(searchQ))
+    : moviesData;
+
+  return (
+    <>
+      <SearchDiv>
+        <SearchStyle>
+          <Input
+            id="search"
+            placeholder="Enter the title..."
+            onChange={(event) => setSearchQ(event.target.value)}
+            value={searchQ}
+          />
+        </SearchStyle>
+      </SearchDiv>
+      <WrapperGrid>
+        {filteredList.map((film, id) => (
+          <MovieCardWrapper key={id}>
```

Comment:
Calling an index an `id` doesn't stop it from being the array's index ;)
```suggestion
        {filteredList.map((film) => (
          <MovieCardWrapper key={film.id}>
```

Reply:
ok, now I see, I was trying to put `film.id` in `{filteredList.map((film, film.id) => (` and it said there's a mistaken syntax... I thought I have to pass it first as a parameter before putting in a key. My bad.